storage/range_tree.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram.gruneir@gmail.com)
+
+package storage
+
+import (
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util"
+)
+
+type (
+	// RangeTree represents holds the metadata about the range tree.  There is only
+	// one for the whole cluster.
+	RangeTree proto.RangeTree
+	// RangeTreeNode represents a node in the range tree, each range
+	// has only a single node.
+	RangeTreeNode proto.RangeTreeNode
+)
+
+// SetupRangeTree creates a new range tree.  This should only be called as part of
+// store.BootstrapRange.
+func (s *Store) SetupRangeTree(batch engine.Engine, ms *engine.MVCCStats,
+	timestamp proto.Timestamp, raftID int64) (*proto.RangeTree, *proto.RangeTreeNode, error) {
+	var previousTree proto.RangeTree
+	ok, err := engine.MVCCGetProto(batch, engine.KeyRangeTreeRoot, timestamp, nil, &previousTree)
+	if err != nil {
+		return nil, nil, err
+	}
+	if ok {
+		return nil, nil, util.Errorf("range tree already exists")
+	}
+
+	tree := &proto.RangeTree{
+		RootID: &raftID,
```

Comment:
Is there a reason that `RootID` has to be nullable? Seems like we'll always have at least 1 range so `RootID` should always have a value.